spec/javascript/availability/components/ill_link.jsx.spec.js

```diff
@@ -0,0 +1,194 @@
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&aulast=author%20name&date=2021');
+});
+
+test('renders an archival thesis link', async () => {
+  const {getByRole, container} = render(
+    <IllLink holding={{locationID: 'ARKTHESES', callNumber: '123'}} />
+  );
+
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent('Request Scan - Penn State Users');
+
+  expect(getByRole('link')).toHaveAttribute('href',
+    'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?Action=10' +
+    '&Form=20&Genre=GenericRequestThesisDigitization' + // archival thesis params
+    '&title=book%20title&callno=123' +
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&aulast=author%20name&date=2021');
+});
+
+test('renders a link with no author info', async () => {
+  $.get = jest.fn().mockImplementation(() => {
+    return Promise.resolve({
+      title_245ab_tsim: 'book title',
+      pub_date_illiad_ssm: 2021,
+      isbn_ssm: '1234'
+    });
+  });
+
+  const {getByRole, container} = render(
+    <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+  );
+
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent('Copy unavailable, request via Interlibrary Loan');
+
+  expect(getByRole('link')).toHaveAttribute('href',
+    'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?' +
+    'Action=10&Form=30&isbn=1234&title=book%20title&callno=123' +
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&date=2021');
+});
+
+test('renders a link with no publication date', async () => {
+  $.get = jest.fn().mockImplementation(() => {
+    return Promise.resolve({
+      title_245ab_tsim: 'book title',
+      author_tsim: 'author name',
+      isbn_ssm: '1234'
+    });
+  });
+
+  const {getByRole, container} = render(
+    <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+  );
+
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent('Copy unavailable, request via Interlibrary Loan');
+
+  expect(getByRole('link')).toHaveAttribute('href',
+    'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?' +
+    'Action=10&Form=30&isbn=1234&title=book%20title&callno=123' +
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&aulast=author%20name');
```

Comment:
The link test looks really repetitious. Would it be easier to have a helper function that you could pass the parameters to and its sends back the string? You can then bake in the "https://" stuff. There might be some handy JS url builders that would make easier on the eyes.... just a thought though.

Reply:
I agree. There is a lot of repetition here since all of the tests are really similar. I'll clean this up so there isn't so much code duplication.